tfhe/src/core_crypto/algorithms/lwe_keyswitch.rs

```diff
@@ -417,12 +417,12 @@ pub fn par_keyswitch_lwe_ciphertext_with_thread_count<Scalar, KSKCont, InputCont
         lwe_keyswitch_key.decomposition_level_count(),
     );
 
-    let mut intermediate_accumulators: Vec<_> = Vec::with_capacity(thread_count.0);
+    // Don't go above the current number of threads
+    let thread_count = thread_count.0.min(rayon::current_num_threads());
+    let mut intermediate_accumulators: Vec<_> = Vec::with_capacity(thread_count);
```

Comment:
nitpick: the `: Vec<_>` anotation should be removed as the function used is `Vec::` and the annotation does not precise the type.

Reply:
indeed, that looks silly